routing_common/bicycle_model.cpp

```diff
@@ -173,7 +173,10 @@ VehicleModel::SurfaceInitList const kBicycleSurface = {
   {{"psurface", "unpaved_good"}, {1.0, 1.0}},
   {{"psurface", "unpaved_bad"}, {0.3, 0.3}},
   // no dedicated cycleway, doesn't mean that bicycle is not allowed, just lower weight
-  {{"hwtag", "nocycleway"}, {0.8, 0.8}},
+  // But why? If nocycleway is tagged explicitly means there is no cycling infra for sure.
```

Comment:
maybe it was assumed that "no cycleway here" will be more often tagged in places which were expected to have cycleway like busy road?

but it is fairly bad assumption IMHO

(or maybe reason was different? is there any cycleway that is not getting bonus but would be not hit by this penalty? like sharrows? but getting bonus to sharrows more likely to be on major busy roads would be also silly)

Reply:
There is some context in https://github.com/organicmaps/organicmaps/pull/5192#discussion_r1826000716 and https://github.com/orgs/organicmaps/discussions/5158#discussioncomment-5938807

I found several cases where the 0.8 factor is contributing to weird detours.

I'm not 100% sure its better to remove the adjustment completely though. Maybe a small factor like 0.95 is beneficial.

Its like assuming that e.g. an explicit `smoothness=good surface=asphalt` footway is a better bet than a footway with no additional tags. WDYT?

Reply:
Is there any known case at all where this extra complexity helps? And this 0.8 is a massive penalty.

(61.5445696, 23.9394003) -> (61.6153965, 23.876755) was mentioned, I guess it is https://www.openstreetmap.org/directions?engine=fossgis_osrm_bike&route=61.54%2C23.94%3B61.62%2C23.88#map=13/61.57544/23.89835&layers=N

but it surely can be achieved by looking at `highway=secondary maxspeed=80` and `highway=tertiary maxspeed=50`, not at effectively random "where missing cycleway was mapped"